sqlx-mysql/src/connection/executor.rs

```diff
@@ -183,7 +183,11 @@ impl MySqlConnection {
             loop {
                 // query response is a meta-packet which may be one of:
                 //  Ok, Err, ResultSet, or (unhandled) LocalInfileRequest
-                let mut packet = self.inner.stream.recv_packet().await?;
+                let mut packet = self.inner.stream.recv_packet().await.inspect_err(|_| {
+                    // if a prepared statement vanished on the server side, we get an error here
```

Comment:
It would be good to link to the specific issue for context, if you created one (I didn't look).

Reply:
This PR is also fine.